pkg/apis/core/v1alpha1/uiresource_types.go

```diff
@@ -115,6 +115,8 @@ type DisableResourceStatus struct {
 	// How many of the resource's sources are disabled.
 	DisabledCount int32 `json:"disabledCount" protobuf:"varint,2,opt,name=disabledCount"`
 
+	PendingCount int32 `json:"pendingCount" protobuf:"varint,4,opt,name=pendingCount"`
```

Comment:
what do you think of having a single State enum that uses the same Disabled | Pending | Enabled | Error you described above, then using that in DisableResourceStatus and DisableStatus and ManifestState?

it feels weird to have so many different ways to represent the same state (in this PR alone we have: a DisableResult enum, a EnableStatus enum, a EnabledCount/DisabledCount field, and an IsDisabled boolean)

Reply:
yeah, that seems better